packages/babel-plugin-proposal-decorators/src/index.js

```diff
@@ -215,32 +215,29 @@ export default function() {
     inherits: syntaxDecorators,
 
     visitor: {
-      ExportDefaultDeclaration(path) {
-        if (!path.get("declaration").isClassDeclaration()) return;
-
-        const { node } = path;
-        const ref =
-          node.declaration.id || path.scope.generateUidIdentifier("default");
-        node.declaration.id = ref;
-
-        // Split the class declaration and the export into two separate statements.
-        path.replaceWith(node.declaration);
-        path.insertAfter(
-          t.exportNamedDeclaration(null, [
-            t.exportSpecifier(ref, t.identifier("default")),
-          ]),
-        );
-      },
       ClassDeclaration(path) {
         const { node } = path;
 
+        if (!hasClassDecorators(node) && !hasMethodDecorators(node.body.body)) {
+          return;
+        }
+
         const ref = node.id || path.scope.generateUidIdentifier("class");
+        const letDeclaration = t.variableDeclaration("let", [
+          t.variableDeclarator(ref, t.toExpression(node)),
+        ]);
 
-        path.replaceWith(
-          t.variableDeclaration("let", [
-            t.variableDeclarator(ref, t.toExpression(node)),
-          ]),
-        );
+        if (path.parentPath.isExportDefaultDeclaration()) {
+          // Split the class declaration and the export into two separate statements.
+          path.parentPath.replaceWithMultiple([
+            letDeclaration,
+            t.exportNamedDeclaration(null, [
+              t.exportSpecifier(ref, t.identifier("default")),
+            ]),
+          ]);
```

Comment:
TODO: Use `babel-helper-split-export-declaration` when https://github.com/babel/babel/pull/6963 is merged